src/segment-loader.js

```diff
@@ -1231,22 +1235,47 @@ export default class SegmentLoader extends videojs.EventTarget {
     // can be adjusted by the timestamp offset
     if (!segmentInfo.hasAppendedData_) {
       this.metadataQueue_.caption.push(
-        this.handleCaptions_.bind(this, simpleSegment, captions));
+        this.handleCaptions_.bind(this, simpleSegment, captionData));
       return;
     }
 
     const timestampOffset = this.sourceUpdater_.videoTimestampOffset() === null ?
       this.sourceUpdater_.audioTimestampOffset() :
       this.sourceUpdater_.videoTimestampOffset();
 
-    captions.forEach((caption) => {
-      createCaptionsTrackIfNotExists(
-        this.inbandTextTracks_, this.hls_.tech_, caption.stream);
-      addCaptionData({
-        captionArray: [caption],
-        inbandTextTracks: this.inbandTextTracks_,
-        timestampOffset
-      });
+    const tracks = {};
```

Comment:
maybe it would be less confusing to call this `captionTracks` so it's less likely to be confused with video/audio tracks

Reply:
done